lit_tests/kernel/wave/codegen.py

```diff
@@ -691,7 +691,8 @@ def test_reduce_max():
         )
     ]
     constraints += [tkw.WorkgroupConstraint(M, BLOCK_M, 1)]
-    constraints += [tkw.WorkgroupConstraint(N, BLOCK_N, 0)]
+    constraints += [tkw.WorkgroupConstraint(N, N, 0)]
+    constraints += [tkw.TilingConstraint(N, BLOCK_N)]
```

Comment:
Does this work if you remove 
```
constraints += [tkw.WorkgroupConstraint(N, N, 0)]
```
? Or is this required to get the workgroup_id.x ?

Reply:
we need it to get workgroup_id.x yes